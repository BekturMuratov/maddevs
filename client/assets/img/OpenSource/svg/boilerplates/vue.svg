<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>