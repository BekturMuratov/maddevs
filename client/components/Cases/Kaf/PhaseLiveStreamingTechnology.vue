<template>
  <section class="container_regular">
    <h2 class="case_title_h2 m-24_bottom media-m-20_bottom">Phase 2: Live streaming technology</h2>
    <TextParagraph class="m-48_bottom media-m-20_bottom">
      To provide viewers with access to their favourite TV channels, the platform needed to emulate the TV experience.
      To add live-streaming technology, Mad Devs leveraged its existing open-source project, Yourcast.tv.
    </TextParagraph>
    <h3 class="case_title_h3">TV shows & TV series</h3>
    <TextQuoteBox class="m-auto m-48_top m-48_bottom media-m-24_top media-m-24_bottom">
      Yourcast.tv by Mad Devs is used to create TV-like streams using YouTube video content. All you need is to create a
      list of YouTube videos that you want to stream. The streaming starts a few seconds after you save your list of
      videos.
      <span class="case_blockquote-box_text-modificator"> Read more about Yourcast.TV. </span>
    </TextQuoteBox>
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      The customer purchased access to several popular TV channels streamed online, and Mad Devs added additional code
      to start re-streaming the channels using Yourcast.TV. As a result, the platform’s viewers received a selection of
      TV channel streams on their devices. HBO, Showtime and other TV channels were seamlessly added to the list of
      available options.
    </TextParagraph>
    <TextParagraph class="m-48_bottom">
      Technically, Django was used for storing stream entries for the Live TV service. The admin created the entries
      that the platform’s Live TV service then pulled down from Django.
    </TextParagraph>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/TextParagraph'
import TextQuoteBox from '@/components/Cases/TextQuoteBox'

export default {
  name: 'PhaseLiveStreamingTechnology',
  components: {
    TextParagraph,
    TextQuoteBox,
  },
}
</script>
