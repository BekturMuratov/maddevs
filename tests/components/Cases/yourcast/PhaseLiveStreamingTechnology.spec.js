import PhaseLiveStreamingTechnology from '@/components/Cases/yourcast/PhaseLiveStreamingTechnology'
import { render, screen } from '@testing-library/vue'
import '../../../__mocks__/intersectionObserverMock'

describe('PhaseLiveStreamingTechnology component', () => {
  it('should render correctly', () => {
    const { container } = render(PhaseLiveStreamingTechnology, {
      mocks: {
        getMediaFromS3: () => 'img.jpg',
      },
    })
    expect(container).toMatchSnapshot()
  })

  it('title should include classname', () => {
    render(PhaseLiveStreamingTechnology, {
      mocks: {
        getMediaFromS3: () => 'img.jpg',
      },
    })
    expect(screen.getByText(/Phase 2: Live streaming technology/i).className).toContain('case_title_h2')
  })
})
