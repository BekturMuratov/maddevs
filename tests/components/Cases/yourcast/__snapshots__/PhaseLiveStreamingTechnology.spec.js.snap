// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseLiveStreamingTechnology component should render correctly 1`] = `
<div>
  <section
    class="live-streaming-technology"
  >
    <div
      class="container_regular"
    >
      <h3
        class="case_title_h2 m-24_bottom media-m-20_bottom"
      >
        
      Phase 2: Live streaming technology
    
      </h3>
       
      <p
        class="case_paragraph m-48_bottom media-m-20_bottom"
      >
        
      To provide viewers with access to their favourite TV channels, the platform needed to emulate the TV experience.
      To add live-streaming technology, Mad Devs leveraged its existing open-source project, Yourcast.tv.
    
      </p>
    </div>
     
    <div
      class="container_full background-color-black-primary case_video-wrapp"
    >
      <video
        class="case_yourcast-tv"
        id="yourcast-tv"
        loop="loop"
        muted="muted"
      >
        <source
          src="img.jpg"
          type="video/mp4"
        />
        
      Your browser does not support the video tag.
    
      </video>
    </div>
     
    <div
      class="container_regular case_posters-wrapp m-60_bottom media-m-48_bottom"
    >
      <ul
        class="case_posters"
      >
        <p
          class="case_posters-title"
        >
          
        Posters:
      
        </p>
         
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:African_Apocalypse_Poster.jpg"
            target="_blank"
          >
            
          African Apocalypse,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:Gasmann_-_Plakat_.jpg"
            target="_blank"
          >
            
          Gasman,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:Help2021_poster.jpg"
            target="_blank"
          >
            
          Help,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:GENESIS_2.0.jpg"
            target="_blank"
          >
            
          Genesis 2.0,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:Alberto_and_the_Concrete_Jungle_retro_travel_poster.jpg"
            target="_blank"
          >
            
          Alberto and the Concrete Jungle,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:Black_Ruby_Poster.jpg"
            target="_blank"
          >
            
          Black Ruby,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="hhttps://commons.wikimedia.org/wiki/File:Creditors_Film_Poster.png"
            target="_blank"
          >
            
          Creditors Film,
        
          </a>
        </li>
        <li
          class="case_poster"
        >
          <a
            class="case_poster-link"
            href="https://commons.wikimedia.org/wiki/File:FINAL_LEGIONARIOsmall.jpg"
            target="_blank"
          >
            
          FINAL LEGIONARIO
        
          </a>
        </li>
      </ul>
    </div>
     
    <div
      class="container_regular"
    >
      <h4
        class="case_title_h3"
      >
        
      What is Yourcast.tv?
    
      </h4>
       
      <blockquote
        class="case_blockquote-box m-auto m-48_top m-48_bottom media-m-24_top media-m-24_bottom"
      >
        
      Yourcast.tv by Mad Devs is used to create TV-like streams using YouTube video content. All you need is to create a
      list of YouTube videos that you want to stream. The streaming starts a few seconds after you save your list of
      videos.
      
        <span
          class="case_blockquote-box_text-modificator"
        >
           Read more about Yourcast.TV. 
        </span>
      </blockquote>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      The customer purchased access to several popular TV channels streamed online, and Mad Devs added additional code
      to start re-streaming the channels using Yourcast.TV. As a result, the platform’s viewers received a selection of
      TV channel streams on their devices. HBO, Showtime and other TV channels were seamlessly added to the list of
      available options.
    
      </p>
       
      <p
        class="case_paragraph m-48_bottom"
      >
        
      Technically, Django was used for storing stream entries for the Live TV service. The admin created the entries
      that the platform’s Live TV service then pulled down from Django.
    
      </p>
    </div>
     
    <div
      class="container_full background-color-silver m-96_bottom media-m-48_bottom"
    >
      <div
        class="container_regular"
      >
        <div
          class="case_lottie"
          id="streaming-technology"
          style="width: 100%; height: 100%; overflow: hidden; margin: 0px auto;"
        >
          <svg
            height="448"
            preserveAspectRatio="xMidYMid meet"
            style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
            viewBox="0 0 519 448"
            width="519"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <clippath
                id="__lottie_element_2"
              >
                <rect
                  height="448"
                  width="519"
                  x="0"
                  y="0"
                />
              </clippath>
              <clippath
                id="__lottie_element_37"
              >
                <path
                  d="M0,0 L519,0 L519,448 L0,448z"
                />
              </clippath>
              <clippath
                id="__lottie_element_80"
              >
                <path
                  d="M0,0 L519,0 L519,448 L0,448z"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#__lottie_element_2)"
            >
              <g
                opacity="1"
                style="display: block;"
                transform="matrix(1,0,0,1,-0.75,-0.75)"
              >
                <g
                  opacity="1"
                  transform="matrix(1,0,0,1,113.75,62.25)"
                >
                  <path
                    d="M0 0"
                    fill-opacity="0"
                    stroke="rgb(115,48,197)"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                    stroke-opacity="1"
                    stroke-width="1.5"
                  />
                </g>
              </g>
              <g
                clip-path="url(#__lottie_element_80)"
                style="display: none;"
              >
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                opacity="1"
                style="display: block;"
                transform="matrix(1,0,0,1,292.25,-0.75)"
              >
                <g
                  opacity="1"
                  transform="matrix(1,0,0,1,113.75,62.25)"
                >
                  <path
                    d="M0 0"
                    fill-opacity="0"
                    stroke="rgb(115,48,197)"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                    stroke-opacity="1"
                    stroke-width="1.5"
                  />
                </g>
              </g>
              <g
                clip-path="url(#__lottie_element_37)"
                style="display: none;"
              >
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
                <g
                  style="display: none;"
                >
                  <g>
                    <path />
                  </g>
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path
                    fill-opacity="0"
                    stroke-linecap="butt"
                    stroke-linejoin="miter"
                    stroke-miterlimit="4"
                  />
                </g>
              </g>
              <g
                style="display: none;"
              >
                <g>
                  <path />
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </section>
</div>
`;
